tests/unit/s2n_ktls_test.c

```diff
@@ -13,14 +13,58 @@
  * permissions and limitations under the License.
  */
 
-#include "crypto/s2n_cipher.h"
+#include "tls/s2n_ktls.h"
+
 #include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
+
+/* set kTLS supported cipher */
+struct s2n_cipher ktls_temp_supported_cipher = {
+    .ktls_supported = true,
+};
+struct s2n_record_algorithm ktls_temp_supported_record_alg = {
+    .cipher = &ktls_temp_supported_cipher,
+};
+struct s2n_cipher_suite ktls_temp_supported_cipher_suite = {
+    .record_alg = &ktls_temp_supported_record_alg,
+};
+
+S2N_RESULT s2n_test_configure_valid_ktls_connection(struct s2n_connection *conn)
```

Comment:
https://github.com/aws/s2n-tls/pull/4066#discussion_r1260340132 I really think this shouldn't be "configure ktls connection". You're not actually configuring ktls. You're configuring a basic tls connection so that ktls can later be enabled.